examples/solvers/find_optimal_parameters.py

```diff
@@ -0,0 +1,234 @@
+"""
+Example of how to use optimization in order to pick reconstruction parameters.
+
+In this example, we solve the tomographic inversion problem with different
+regularizers (fbp, huber and tv) and pick the "best" regularization parameter
+for each method w.r.t. a set of reference data.
+
+To find the "best" parameter we use Powells method to optimize a figure of
+merit, here the L2-distance to the true result.
+"""
+
+import numpy as np
+import odl
+import scipy
+
+
+def optimal_parameters(reconstruction, fom, phantoms, data,
+                       initial_param=0):
+    """Find the optimal parameters for a reconstruction method.
+
+    Notes
+    -----
+    For a forward operator :math:`A : X \to Y`, a reconstruction operator
+     parametrized by :math:`\theta` is some operator :math:`R_\theta : Y \to X`
+     such that
+
+    .. math::
+        R_\theta(A(x)) \approx x.
+
+    The optimal choice of :math:`\theta` is given by
+
+    .. math::
+        \theta = \argmin_\theta fom(R(A(x) + noise), x)
+
+    where :math:`fom : X \times X \to \mathbb{R}` is a Figure of Merit.
+
+
+    Parameters
+    ----------
+    reconstruction : callable
+        Function that takes two parameters:
+
+            * data : The data to be reconstructed
+            * parameters : Parameters of the reconstruction method
+
+        The function should return the reconstructed image.
+    fom : callable
+        Function that takes two parameters:
+
+            * reconstructed_image
+            * true_image
+
+        and returns a scalar Figure of Merit.
+    phantoms : sequence
+        True images.
+    data : sequence
+        The data to be reconstructed.
+    initial_param : array-like
+        Initial guess for the parameters.
+    """
+
+    def func(lam):
+        # Function to be minimized by scipy
+        return sum(fom(reconstruction(datai, lam), phantomi)
+                   for phantomi, datai in zip(phantoms, data))
+
+    # Pick resolution to fit the one used by the space
+    tol= np.finfo(phantoms[0].space.dtype).resolution * 10
+
+    initial_param = np.asarray(initial_param)
+
+    if initial_param.size == 1:
```

Comment:
Add a word on what is happening here (we use a faster optimizer for the one parameter case).